extensions/reactive-pg-client/deployment/src/main/java/io/quarkus/reactive/pg/client/deployment/ReactivePgClientProcessor.java

```diff
@@ -121,6 +123,11 @@ void addHealthCheck(
                         dataSourcesBuildTimeConfig.healthEnabled));
     }
 
+    @BuildStep
+    void unremoveableBeans(BuildProducer<UnremovableBeanBuildItem> producer) {
+        producer.produce(UnremovableBeanBuildItem.beanTypes(PgPoolCreator.class));
```

Comment:
For my own education what happens when:
- like in most cases, the user doesn't provide an instance?
- the user provides more than one instance?

Reply:
Currently, it only works if there is exactly one instance. In all other cases, nothing happens.

We should definitely log a warning (or maybe even fail) if there are multiple instances

Reply:
Right, it only gets created once (if at all)

Reply:
It is now a build time error if more than one beans of `PgPoolCreator` exist